ts/screens/messages/paginated/MessageRouterScreen.tsx

```diff
@@ -0,0 +1,172 @@
+import { useNavigationContext } from "../../../utils/hooks/useOnFocus";
+
+type OwnProps = NavigationInjectedProps<{ messageId: UIMessageId }>;
+type Props = OwnProps &
+  ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps>;
+
+/**
+ * Choose the screen where to navigate, based on the message content.
+ * TODO: refactor to return an action for _dispatch_
+ * @param message
+ * @param messageDetails
+ */
+const navigateToScreenHandler =
+  (message: UIMessage, messageDetails: UIMessageDetails) =>
+  (dispatch: OwnProps["navigation"]["dispatch"]) => {
+    if (euCovidCertificateEnabled && messageDetails.euCovidCertificate) {
+      navigateBack();
+      navigateToEuCovidCertificateDetailScreen({
+        authCode: messageDetails.euCovidCertificate
+          .authCode as EUCovidCertificateAuthCode,
+        messageId: message.id
+      });
+    } else {
+      navigateBack();
+      dispatch(
+        navigateToPaginatedMessageDetailScreenAction({
+          message,
+          messageDetails
+        })
+      );
+    }
+  };
+
+const MessageRouterScreen = ({
+  cancel,
+  cursors,
+  loadMessageDetails,
+  loadPreviousPage,
+  reloadPage,
+  maybeMessage,
+  maybeMessageDetails,
+  messageId
+}: Props): React.ReactElement => {
+  const navigation = useNavigationContext();
+  // used to automatically dispatch loadMessages if the pot is not some at the first rendering
+  // (avoid displaying error at the first frame)
+  const firstRendering = useRef(true);
+  const isLoading = !pot.isError(maybeMessageDetails);
+
+  const tryLoadMessageDetails = useCallback(() => {
+    if (maybeMessage === undefined) {
+      if (pot.isNone(cursors)) {
+        // nothing in the collection, refresh
+        reloadPage();
+      } else if (pot.isSome(cursors)) {
+        // something in the collection, get the new ones only
+        loadPreviousPage(cursors.value.previous);
+      }
+    }
+    loadMessageDetails(messageId);
+  }, [
+    maybeMessage,
+    cursors,
+    messageId,
+    loadMessageDetails,
+    loadPreviousPage,
+    reloadPage
+  ]);
+
+  useEffect(() => {
+    // message in the list and its details loaded: green light
+    if (isStrictSome(maybeMessageDetails) && maybeMessage !== undefined) {
+      navigateToScreenHandler(
+        maybeMessage,
+        maybeMessageDetails.value
+      )(navigation.dispatch);
+      return;
+    }
+    if (firstRendering.current) {
+      tryLoadMessageDetails();
+      // eslint-disable-next-line functional/immutable-data
+      firstRendering.current = false;
+    }
+  }, [
+    loadMessageDetails,
+    maybeMessage,
+    maybeMessageDetails,
+    messageId,
+    navigation,
+    tryLoadMessageDetails
+  ]);
+
+  return (
+    <LoadingErrorComponent
+      errorText={I18n.t("global.genericError")}
+      isLoading={isLoading}
+      loadingCaption={I18n.t("messageDetails.loadingText")}
+      onAbort={cancel}
+      onRetry={tryLoadMessageDetails}
+    />
```

Comment:
In order to support the contextual help (top-right) and back (top-left) we should wrap this component in the `BaseScreenComponent`

```suggestion
    <BaseScreenComponent goBack={true} contextualHelp={emptyContextualHelp}>
      <LoadingErrorComponent
        errorText={I18n.t("global.genericError")}
        isLoading={isLoading}
        loadingCaption={I18n.t("messageDetails.loadingText")}
        onAbort={cancel}
        onRetry={tryLoadMessageDetails}
      />
    </BaseScreenComponent>
```

| now  | suggestion |
| ------------- | ------------- |
![now](https://user-images.githubusercontent.com/822471/147258057-96c5a97e-5936-457b-ba89-2a6a893f9668.png) | ![suggestion](https://user-images.githubusercontent.com/822471/147258059-efd30f9d-706a-4b6a-9a7a-d14e69471e0b.png)

what do you think?

Reply:
Great catch! Fixed here https://github.com/pagopa/io-app/pull/3580/commits/7ed5821c4ef577460ebac6a7a0e8ce7109435060